### PR TITLE
SIL: Fix opaque type erasure after clearing [serialized] flags

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://problem/115355709.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: test/SIL/Serialization/opaque_return_type_serialize.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module %s
```
Comment (nate-chandler) [inline test/SIL/Serialization/opaque_return_type_serialize.swift:2]: Looks like this needs `-disable-availability-checking`.

**Review by nate-chandler — APPROVED** [review]

**Comment by aschwaighofer** [comment]
Thank you for fixing this

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**